addons/purchase/models/account_invoice.py

```diff
@@ -36,7 +36,7 @@ def _prepare_invoice_line_from_po_line(self, line):
         else:
             qty = line.qty_received - line.qty_invoiced
         if float_compare(qty, 0.0, precision_rounding=line.product_uom.rounding) <= 0:
-            qty = 0.0
+            qty = qty
```

Comment:
For purchase refund case: 
1) when we receieved all products  and we generate and validate/paid invoice, then we return some products with refund.  A new invoice should be generated with negative value for returned products.  

so for this case we have to change qty..